cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -133,6 +133,8 @@ macro(build_dependency DEPENDENCY_NAME)
     build_grpc()
   elseif("${DEPENDENCY_NAME}" STREQUAL "BZip2")
     build_bzip2()
+  elseif("${DEPENDENCY_NAME}" STREQUAL "uriparser")
+    build_uriparser()
```

Comment:
@xhochy I'm not sure why it's necessary to hardcode all these paths?

Reply:
It's not possible in CMake to do construct function names at runtime, i.e. it has no `eval` functionality.